lib/gui/components/modal/services/modal.js

```diff
@@ -56,18 +56,32 @@ module.exports = function($uibModal, $q) {
       close: modal.close,
       result: $q((resolve, reject) => {
         modal.result
-          .then(resolve)
-          .catch((error) => {
+          .then((value) => {
+            AnalyticsService.logEvent('Modal closed', {
```

Comment:
This event feels weird. This is what I get if I enable a dangerous setting:

```
tracker.js:10 Mon Feb 20 2017 16:17:53 GMT-0400 (BOT) Dangerous settings modal ({"setting":"unsafeMode"})
tracker.js:10 Mon Feb 20 2017 16:17:54 GMT-0400 (BOT) Modal closed ({"method":"User accepted","value":true})
```

Let's mimic what we do for the other settings:

```
Set setting ({"setting":"errorReporting","value":true})
```

Maybe simply replacing `Set setting` with `Set dangerous setting`.